pkg/registry/apis/dashboard/register.go

```diff
@@ -144,42 +183,347 @@ func (b *DashboardsAPIBuilder) InstallSchema(scheme *runtime.Scheme) error {
+			return err
+		}
+	}
+
+	// Validate refresh interval
+	if err := b.validateRefreshInterval(refresh); err != nil {
+		return err
+	}
+
+	id, err := identity.GetRequester(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting requester: %w", err)
+	}
+
+	internalId, err := id.GetInternalID()
+	if err != nil {
+		return fmt.Errorf("error getting internal ID: %w", err)
+	}
+
+	// Validate quota
+	if !a.IsDryRun() {
+		params := &quota.ScopeParameters{}
+		params.OrgID = id.GetOrgID()
+		params.UserID = internalId
+
+		quotaReached, err := b.QuotaService.CheckQuotaReached(ctx, dashboards.QuotaTargetSrv, params)
+		if err != nil {
+			// Ignore quota disabled errors
+			// TODO: Research
+			if !strings.Contains(err.Error(), "quota.disabled") {
+				return fmt.Errorf("error checking quota: %w", err)
 			}
 		}
+		if quotaReached {
+			return fmt.Errorf("dashboard quota reached") // TODO: Add a more specific error message as before
+		}
+	}
+
+	return nil
+}
+
+// validateUpdate validates dashboard updates
+func (b *DashboardsAPIBuilder) validateUpdate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error {
+	// Get the new and old dashboards
+	newDashObj := a.GetObject()
+	oldDashObj := a.GetOldObject()
+
+	title, refresh, err := getDashboardProperties(newDashObj)
+	if err != nil {
+		return fmt.Errorf("error extracting dashboard properties: %w", err)
+	}
+
+	oldAccessor, err := utils.MetaAccessor(oldDashObj)
+	if err != nil {
+		return fmt.Errorf("error getting meta accessor: %w", err)
+	}
+
+	newAccessor, err := utils.MetaAccessor(newDashObj)
+	if err != nil {
+		return fmt.Errorf("error getting meta accessor: %w", err)
+	}
+
+	// Parse namespace for old dashboard
+	nsInfo, err := types.ParseNamespace(oldAccessor.GetNamespace())
+	if err != nil {
+		return fmt.Errorf("failed to parse namespace: %w", err)
+	}
+
+	// Validate requester's organization context
+	id, err := identity.GetRequester(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting requester: %w", err)
+	}
+
+	// Validate organization access
+	if id.GetOrgID() != nsInfo.OrgID {
+		return apierrors.NewNotFound(a.GetResource().GroupResource(), a.GetName())
+	}
+
+	// Basic validations
+	if err := b.validateBasicProperties(title, newAccessor); err != nil {
+		return err
+	}
+
+	// Validate folder existence if specified and changed
+	if !a.IsDryRun() && newAccessor.GetFolder() != "" && newAccessor.GetFolder() != oldAccessor.GetFolder() {
+		if err := b.validateFolderExists(ctx, newAccessor.GetFolder(), nsInfo.OrgID); err != nil {
+			return err
+		}
+	}
+
+	// Validate refresh interval
+	if err := b.validateRefreshInterval(refresh); err != nil {
+		return err
+	}
+
+	// Check for provisioning - disallow updates to provisioned dashboards if not allowed
+	if err := b.validateProvisionedDashboardUpdate(ctx, oldAccessor); err != nil {
+		return err
+	}
+
+	// Validate generation conflicts
+	if oldAccessor.GetGeneration() > newAccessor.GetGeneration() {
```

Comment:
should this be:
```suggestion
	if oldAccessor.GetGeneration() != newAccessor.GetGeneration() {
```
below i see this check:
```
		// check for is someone else has written in between
		if dash.Version != existing.Version {
			if cmd.Overwrite {
				dash.SetVersion(existing.Version)
			} else {
				return dashboards.ErrDashboardVersionMismatch
			}
		}
```

Reply:
Good catch, thank you!